### ISSUE TITLE
[Scaffolder] RepoUrlPicker allowedOwners selection changes automatically

### ISSUE THREAD

**Issue by cklewin** [backstage/backstage#13730]
## Expected Behavior

When multiple allowedOwners are in the template, the user selects the desired owner and the selection remains while filling out the rest of the template form.

## Actual Behavior

When multiple allowedOwners exist, the user has to select the desired owner twice to get the selection to "stick", and also the selection switches back to the first option when the "Repository" field is updated.

## Steps to Reproduce

1. Use a template with RepoUrlPicker containing multiple allowedOwners
    ```
    - title: Choose a location
      required:
        - repoUrl
      properties:
        repoUrl:
          title: Repository Location
          type: string
          ui:field: RepoUrlPicker
          ui:options:
            allowedHosts:
              - github.com
            allowedOwners:
              - some-github-owner-A
              - some-github-owner-B
              - some-github-owner-C
            requestUserCredentials:
              secretsKey: USER_OAUTH_TOKEN
    ```
2. Select an owner (B) other than the first (A), it should remain on the first (A)
3. Select owner (B) again, it should now remain selected (B)
4. Start typing in the Repository field, the owner should switch back to the first (A)

## Context

We're trying to limit the owners available to the user by presenting a list of GitHub organizations to choose from.

This results in difficulty for the user to select their owner, they have to select the owner multiple times to get it to "stick" when filling out the template forms, or they don't see the owner switch on them and end up creating the repository in the wrong organization.

## Your Environment

This behaviour happens in Safari and Firefox, though I mainly work in Safari.

- Browser Information: Safari Version 15.6.1 (17613.3.9.1.16)

- Output of `yarn backstage-cli info`: 

```text
OS:   Darwin 21.6.0 - darwin/arm64
node: v18.7.0
yarn: 3.2.1
cli:  0.18.1 (installed)
backstage:  1.4.0

Dependencies:
  @backstage/app-defaults                          1.0.5
  @backstage/backend-common                        0.14.1, 0.15.0
  @backstage/backend-plugin-api                    0.1.1
  @backstage/backend-tasks                         0.3.4
  @backstage/catalog-client                        1.0.4
  @backstage/catalog-model                         1.1.0
  @backstage/cli-common                            0.1.9
  @backstage/cli                                   0.18.1
  @backstage/config-loader                         1.1.3
  @backstage/config                                1.0.1
  @backstage/core-app-api                          1.0.5
  @backstage/core-components                       0.10.0, 0.11.0
  @backstage/core-plugin-api                       1.0.5
  @backstage/errors                                1.1.0
  @backstage/integration-react                     1.1.3
  @backstage/integration                           1.3.0
  @backstage/plugin-api-docs                       0.8.8
  @backstage/plugin-auth-backend                   0.15.1
  @backstage/plugin-auth-node                      0.2.4
  @backstage/plugin-catalog-backend-module-github  0.1.6
  @backstage/plugin-catalog-backend                1.3.1
  @backstage/plugin-catalog-common                 1.0.5
  @backstage/plugin-catalog-graph                  0.2.20
  @backstage/plugin-catalog-import                 0.8.11
  @backstage/plugin-catalog-node                   1.0.1
  @backstage/plugin-catalog-react                  1.1.3
  @backstage/plugin-catalog                        1.5.0
  @backstage/plugin-github-actions                 0.5.8
  @backstage/plugin-github-pull-requests-board     0.1.2
  @backstage/plugin-home                           0.4.24
  @backstage/plugin-org                            0.5.8
  @backstage/plugin-permission-common              0.6.3
  @backstage/plugin-permission-node                0.6.4
  @backstage/plugin-permission-react               0.4.4
  @backstage/plugin-proxy-backend                  0.2.29
  @backstage/plugin-scaffolder-backend             1.5.1
  @backstage/plugin-scaffolder-common              1.1.2
  @backstage/plugin-scaffolder                     1.5.0
  @backstage/plugin-search-backend-module-pg       0.3.6
  @backstage/plugin-search-backend-node            1.0.1
  @backstage/plugin-search-backend                 1.0.1
  @backstage/plugin-search-common                  1.0.0
  @backstage/plugin-search-react                   1.0.1
  @backstage/plugin-search                         1.0.1
  @backstage/plugin-sonarqube                      0.3.7
  @backstage/plugin-stack-overflow                 0.1.4
  @backstage/plugin-tech-radar                     0.5.15
  @backstage/plugin-techdocs-backend               1.2.1
  @backstage/plugin-techdocs-module-addons-contrib 1.0.3
  @backstage/plugin-techdocs-node                  1.3.0
  @backstage/plugin-techdocs-react                 1.0.3
  @backstage/plugin-techdocs                       1.3.1
  @backstage/plugin-user-settings                  0.4.7
  @backstage/release-manifests                     0.0.5
  @backstage/test-utils                            1.1.3
  @backstage/theme                                 0.2.16
  @backstage/types                                 1.0.0
  @backstage/version-bridge                        1.0.1
```

**Comment by freben** [comment]
Oh! That sounds like some sort of odd race condition or so. Are you able to have a look at fixing it? 🙏

**Comment by cklewin** [comment]
> Oh! That sounds like some sort of odd race condition or so. Are you able to have a look at fixing it? 🙏

I did take a look at some of the code attempting to understand, but unfortunately I couldn't determine the cause.  I agree, some race condition involving the onchange function along with the useState / useEffect usage.  After spending some time attempting to narrow it down, I wasn't able to fully understand what's being done so I opened this report hoping someone with more experience in the existing flow would make sense of it.
